### PR TITLE
remove experimental extended builds feature

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@smarterclayton ptal/confirm this is what you wanted.  i'll send a note out to the appropriate mailing lists once we merge this. (and update docs of course).

@openshift/devex fyi

**Comment by smarterclayton** [comment]
Approved

**Comment by bparees** [comment]
tests seem borked. let's see if closing/reopening helps.

File: pkg/build/apis/build/v1/types.go [hunk 1]
```diff
@@ -691,22 +691,6 @@ type SourceBuildStrategy struct {
...
-	// to copy binary artifacts created from one build into a separate runtime image.
-	RuntimeImage *kapi.ObjectReference `json:"runtimeImage,omitempty" protobuf:"bytes,7,opt,name=runtimeImage"`
-
-	// runtimeArtifacts specifies a list of source/destination pairs that will be
-	// copied from the builder to the runtime image. sourcePath can be a file or
-	// directory. destinationDir must be a directory. destinationDir can also be
-	// empty or equal to ".", in this case it just refers to the root of WORKDIR.
-	// Deprecated: This feature will be removed in a future release. Use ImageSource
-	// to copy binary artifacts created from one build into a separate runtime image.
-	RuntimeArtifacts []ImageSourcePath `json:"runtimeArtifacts,omitempty" protobuf:"bytes,8,rep,name=runtimeArtifacts"`
```
Comment (jim-minter) [inline pkg/build/apis/build/v1/types.go]: @smarterclayton @bparees if we remove fields 7 and 8 from the protobuf spec, what happens if we add new fields in this struct in the future and reuse those numbers?

**Comment by jim-minter** [comment]
@bparees do we get to pull config.Runtime image out of source-to-image as well?

**Comment by bparees** [comment]
> @bparees do we get to pull config.Runtime image out of source-to-image as well?

unfortunately not.  we did not treat extended builds as "experimental" in the s2i CLI tool.  We can deprecate it and plan to remove it someday, but we can't just yank it.

File: pkg/build/apis/build/v1/types.go (hunk 1 above)
Comment (bparees) [reply pkg/build/apis/build/v1/types.go]: discussed on IRC, will add a comment warning people not to reuse 7+8 until we get an answer on this from @smarterclayton or maybe @liggitt knows.

Comment (liggitt) [reply pkg/build/apis/build/v1/types.go]: you cannot reuse protobuf tags. add a tombstone comment like https://github.com/kubernetes/kubernetes/pull/44900/commits/d45a3423f08c95f285f90b7ee55db1f12ae4f9d2#diff-c5f41a1fd7d92b2773d2d5194a0fb6ffR2633

Comment (bparees) [reply pkg/build/apis/build/v1/types.go]: thanks @liggitt, done.

Comment (liggitt) [reply pkg/build/apis/build/v1/types.go]: same goes for json fields, actually

**Pull request merged**